CONTRIBUTING.md

```diff
@@ -0,0 +1,435 @@
+consuming them.
+
+- API of adaptors, similarly as for UI components, should be possibly explicit
+  and simple. Passing the whole store is usually not a good idea because it is
+  not clear which part of the store's API is really used by the adaptor. Taking
+  plain models with well-specified expected roles is usually a better idea.
+
+- Output of one adaptor (usually model) can be an input of another adaptor. This
+  structure should intuitively reflect the branches in the data flow in the application.
+
+## Storybook
+
+### Features
+
+Storybook is an internal tool that supports rapid development of components in
+isolation, outside the application. It is also a kind of catalog of components from
+which the application is built.
+
+It offers a number of functionalities that improve development:
+
+- hot reloading
+- import paths overloading and `stubs` to isolate from the backend
+- integrated test runner
+- figma integration
+- built-in visual components inspector
+- pages organized in groups
+- bunch of dedicated components like `GenericListView` to create pages quickly
+
+> [!NOTE]
+> The `StatusQ` library also contains this type of utility application, called `Sandbox`,
+> which is currently deprecated as it offers less functionality.
+
+### Running
+
+`Storybook` is a `cmake` project and can be easily opened, compiled and run in
+`QtCreator` from the `storybook/CMakeLists.txt` file.
+
+### Good practices
+
+- Single `page` instantiates a single component and only necessary mocks/test
+  data and auxiliary controls interacting with component's API.
+  
+  It makes the pages simple, easy to maintain and directly presenting the API of
+  a given component.
+
+- Storybook pages are valuable for all types of components - both basic ones
+  (delegates, buttons) and more complex ones, doing complex flows like e.g.
+  funds transfer.
+
+- Storybook (and unit tests) has a mechanism of so-called `stubs` to replace
+  real stores with empty objects, but in a way preserving types. It is needed
+  because in Storybook and unit tests backend's context properties used in real
+  stores are not available. Store's stubs are intended to be empty QtObjects,
+  actual mocking should be done within a page.
+  
+  This approach allows making the dependencies of the component truly visible
+  and explicit in a Storybook page. An alternative approach would be to
+  implement single, shared mocks for every store and share the among pages. But
+  it leads to situation that given mock provides much more than is needed by a
+  single UI component. As a consequence, it is not clear on which part of the
+  store given component depends on. A separate thing is that UI components
+  should not depend on the store at all unless it is really necessary (only top
+  level components covering complex flows).
+
+- A component in Storybook should be functional.
+  
+  The state of a component's Storybook page quite accurately reflects the state
```

Comment:
we could also add something about the Models in Storybook, maybe we should try creating stronger models that cover most use cases instead of creating new ones or ones in respective storybook pages, Ive noticed that somewhere already.. this will also mean testing those same in different pages which will lead to better coverage of behaviours in the app